src/test/java/seedu/address/testutil/ModelStub.java

```diff
@@ -303,5 +304,30 @@ public Calendar getCalendar() {
         throw new AssertionError("This method should not be called.");
     }
 
+    @Override
+    public void addAction(DoableAction<?> action) {
+
+    }
+
+    @Override
+    public boolean canUndo() {
+        return false;
+    }
+
+    @Override
+    public boolean canRedo() {
+        return false;
+    }
+
+    @Override
+    public DoableAction<?> undo() {
+        return null;
+    }
+
+    @Override
+    public DoableAction<?> redo() {
+        return null;
+    }
+
```

Comment:
Maybe follow the top and throw assertion error

Reply:
Though these should not be a problem, idk why